src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -801,22 +801,15 @@ public bool RemoveTransientTransactions(uint256 depositId = null)
             }
         }
 
-        /// <inheritdoc />
-        public IEnumerable<IWithdrawal> GetWithdrawals()
-        {
-            foreach (TransactionData transactionData in this.Wallet.MultiSigAddress.Transactions.OrderByDescending(t => t.CreationTime))
-            {
-                Transaction walletTrx = transactionData.GetFullTransaction(this.network);
-                IWithdrawal withdrawal = this.withdrawalExtractor.ExtractWithdrawalFromTransaction(walletTrx, transactionData.BlockHash, transactionData.BlockHeight ?? 0);
-                if (withdrawal == null)
-                    continue;
 
-                yield return withdrawal;
-            }
+        private OutPoint EarliestOutput(Transaction transaction)
+        {
+            var comparer = Comparer<OutPoint>.Create((x, y) => this.CompareOutpoints(x, y));
+            return transaction.Inputs.Select(i => i.PrevOut).OrderBy(t => t, comparer).FirstOrDefault();
         }
 
         /// <inheritdoc />
-        public List<(Transaction, IWithdrawal)> FindWithdrawalTransactions(uint256 depositId = null)
+        public List<(Transaction, IWithdrawal)> FindWithdrawalTransactions(uint256 depositId = null, bool sort = false)
```

Comment:
Will we ever want to find `FindWithDrawalTransactions` unordered?

Reply:
Yes, it is used in that way.